File: docs-examples/pubsub-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/pubsub/BookAuditListener.kt
package io.micronaut.configuration.lettuce.docs.pubsub

import io.micronaut.configuration.lettuce.pubsub.annotation.MessageChannel
import io.micronaut.configuration.lettuce.pubsub.annotation.RedisListener
import io.micronaut.messaging.annotation.MessageBody

@RedisListener
class BookAuditListener {

    // tag::listener[]
    @MessageChannel(value = "books.audit", exceptionHandler = BookAuditExceptionHandler::class)
    fun receive(@MessageBody event: BookCreated) {
        throw IllegalStateException("Could not audit ${event.title}")
    }
    // end::listener[]
}

File: docs-examples/pubsub-python/src/test/python/micronaut/configuration/lettuce/docs/pubsub/BookAuditListener.py
from typing import Annotated

from micronaut.configuration.lettuce.pubsub.annotation import MessageChannel, RedisListener
from micronaut.messaging.annotation import MessageBody

from .BookAuditExceptionHandler import BookAuditExceptionHandler
from .BookCreated import BookCreated


@RedisListener
class BookAuditListener:

    # tag::listener[]
    @MessageChannel(value="books.audit", exceptionHandler=BookAuditExceptionHandler)
    def receive(self, event: Annotated[BookCreated, MessageBody]) -> None:
        raise RuntimeError(f"Could not audit {event.title}")
    # end::listener[]
